### PR TITLE
Updates for php8

### PR DESCRIPTION
**Pull request by whyisjake**
Let's do this.

### CODE REVIEW

**Comment by GaryJones** [comment]
As per https://github.com/Automattic/fb-instant-articles/actions/runs/3074081777 - the integration/WP tests (as little as they are) are now passing on PHP 7.1-PHP 8.0.

PHP 8.1 is failing on one item that needs fixing in @whyisjake's fork of the Facebook SDK. Jake, please see the tests errors on https://github.com/Automattic/fb-instant-articles/actions/runs/3074081777/jobs/4966631390 and adjust the `strpos()` as needed.

The branch is also passing linting (but now CS, when the full WPCS is added). There were no PHPCompatibility issues for `8.0`.

I'll note that on plugin activation, I saw a PHP Warning: "Warning: Trying to access array offset on value of type null in /Users/gary/Sites/fbia/app/public/wp-content/plugins/fb-instant-articles/facebook-instant-articles.php on line 82", so there will still need to be a manual check and test of functionality because the automated tests are very very far from being comprehensive.

I'll also note that I [renamed the repository](https://github.com/Automattic/fb-instant-articles/issues/1094) to make the tests pass locally and on CI.

**Comment by GaryJones** [comment]
> I saw a PHP Warning: "Warning: Trying to access array offset on value of type null

This PR now includes a commit to fix that PHP Warning.

**Comment by GaryJones** [comment]
> PHP 8.1 is failing on one item that needs fixing in @whyisjake's fork of the Facebook SDK. Jake, please see the tests errors on [Automattic/fb-instant-articles/actions/runs/3074081777/jobs/4966631390](https://github.com/Automattic/fb-instant-articles/actions/runs/3074081777/jobs/4966631390) and adjust the `strpos()` as needed.

This should be fixed by merging https://github.com/whyisjake/facebook-instant-articles-sdk-php/pull/2 and then running the tests here again.

**Comment by whyisjake** [comment]
@GaryJones this is ready for review!

File: readme.txt [hunk 1]
```diff
@@ -17,19 +18,19 @@ With the plugin active, a new menu will be available for you to connect to your
...
+Built-in to the plugin are many [pre-defined transformer rules](https://github.com/Automattic/fb-instant-articles/blob/master/rules-configuration.json) which aims to cover standard WordPress installations. If your WordPress content contains elements which are not covered by the built-in ruleset, you can define your own additional rules to extend or override the defaults in the Settings of this plugin, under: **Plugin Configuration** > **Publishing Settings** > **Custom transformer rules**.
 
 = Access to Instant Articles =
 
 The current criteria for access to Instant Articles are:
 * your Facebook Page must have an established presence of at least 90 days
-* your content adheres to the [Instant Article Policies](https://developers.facebook.com/docs/instant-articles/policy/)
+* your content adheres to the [Instant Article Policies](https://developers.fb.com/docsstant-articles/policy/)
```
Comment (GaryJones) [inline readme.txt:27]: Is this change legit?

File: readme.txt [hunk 2]
```diff
@@ -17,19 +18,19 @@ With the plugin active, a new menu will be available for you to connect to your
...
 
-Before your Instant Articles can be published on Facebook, your feed must undergo an initial review and approval. Facebook requires a minimum number of 10 articles in your feed before being eligible for review. The review process checks that your draft Instant Articles are properly formatted, have content consistency with their mobile web counterparts, and adhere to the [community standards](https://www.facebook.com/communitystandards/) and [content policies](https://www.facebook.com/help/publisher/1348682518563619).
+Before your Instant Articles can be published on Facebook, your feed must undergo an initial review and approval. Facebook requires a minimum number of 10 articles in your feed before being eligible for review. The review process checks that your draft Instant Articles are properly formatted, have content consistency with their mobile web counterparts, and adhere to the [community standards](https://www.fb.com/communitystandardsand [content policies](https://www.fb.com/helpblisher/1348682518563619).
```
Comment (GaryJones) [inline readme.txt:29]: Is this change legit?

File: readme.txt [hunk 3]
```diff
@@ -17,19 +18,19 @@ With the plugin active, a new menu will be available for you to connect to your
...
 
 It's important to note that if you use meta fields to add extra text, images or videos to your Posts, Facebook will expect you to add these to your Instant Articles output too. This plugin includes hooks to help you do that.
 
-[See Facebook's documentation for full details of the submission process.](https://developers.facebook.com/docs/instant-articles)
+[See Facebook's documentation for full details of the submission process.](https://developers.fb.com/docsstant-articles)
```
Comment (GaryJones) [inline readme.txt:33]: Is this change legit?

File: readme.txt [hunk 4]
```diff
@@ -60,15 +61,15 @@ When transforming your post into an Instant Article, this plugin will show warni
...
+This plugin transforms your post into an Instant Article by matching markup in your content to one of the [components available](https://github.com/fb/fb-instantticlesk-php/blob/master/docs/QuickStart.md#transformer-classes) in Instant Articles markup. Although the plugin contains many [built-in rules](https://github.com/Automattic/fb-instant-articles/blob/master/rules-configuration.json) to handle common cases, there may be markup in your post which is not recognized by these existing rules. In this case, you may be required to define some of your own rules. See below for more details about where and how.
```
Comment (GaryJones) [inline readme.txt:64]: This doesn't look legit.

File: readme.txt [hunk 5]
```diff
@@ -60,15 +61,15 @@ When transforming your post into an Instant Article, this plugin will show warni
...
 
 = How do I define my own transformer rules so that content from my site is rendered appropriately in an Instant Article? =
 
-Your custom rules can be defined in the Settings of this plugin, under: **Plugin Configuration** > **Publishing Settings** > **Custom transformer rules**. More detailed instructions about all the options available is documented in the [Custom Transformer Rules](https://github.com/facebook/facebook-instant-articles-sdk-php/blob/master/docs/QuickStart.md#custom-transformer-rules) section of the Facebook Instant Articles SDK.
+Your custom rules can be defined in the Settings of this plugin, under: **Plugin Configuration** > **Publishing Settings** > **Custom transformer rules**. More detailed instructions about all the options available is documented in the [Custom Transformer Rules](https://github.com/fb/fb-instantticlesk-php/blob/master/docs/QuickStart.md#custom-transformer-rules) section of the Facebook Instant Articles SDK.
```
Comment (GaryJones) [inline readme.txt:68]: Nor this one.

**Review by GaryJones — DISMISSED** [review]
Looks like some bad replacements in the readme file.

**Review by GaryJones — APPROVED** [review]
These changes all look good. I'd like to get this out as a beta before bumping version and tagging as a final release. Might also give us chance to add in one or two more items from the 4.3.0 milestone.

**Comment by whyisjake** [comment]
Squash and merge or just merge?

**Pull request merged**